### ISSUE TITLE
Fix warning "warning MSB3026: Could not copy "F:\workspace\_work\1\s\.packages\..."

### ISSUE THREAD

**Issue by dougbu** [dotnet/aspnetcore#22635]
Warning likely introduced when Razor SDK was merged into this repo (#21581). Example is [here](https://dev.azure.com/dnceng/public/_build/results?buildId=674833&view=logs&j=366a1024-3a8a-5d08-6c00-a4b17dce0d38&t=7c220d1f-55ee-5c39-b8a6-7373cb0d28ee&l=626). Full text
``` text
F:\workspace\_work\1\s\.dotnet\sdk\5.0.100-preview.6.20266.3\Microsoft.Common.CurrentVersion.targets(4363,5): warning MSB3026: Could not copy "F:\workspace\_work\1\s\.packages\microsoft.codeanalysis.common\3.4.0\lib\netstandard2.0\ko\Microsoft.CodeAnalysis.resources.dll" to "F:\workspace\_work\1\s\artifacts\bin\Microsoft.AspNetCore.Razor.Tools\Release\netcoreapp3.0\ko\Microsoft.CodeAnalysis.resources.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'F:\workspace\_work\1\s\artifacts\bin\Microsoft.AspNetCore.Razor.Tools\Release\netcoreapp3.0\ko\Microsoft.CodeAnalysis.resources.dll' because it is being used by another process.  [F:\workspace\_work\1\s\src\Razor\Microsoft.AspNetCore.Razor.Tools\src\Microsoft.AspNetCore.Razor.Tools.csproj]
```

**Comment by dougbu** [comment]
Part of #22240

**Comment by dougbu** [comment]
This probably shares a root cause with multiple build **failures** we've hit over the last couple of days. Generally looks like a build race while creating the Razor SDK.

**Comment by wtgodbe** [comment]
Failed the build here: https://dev.azure.com/dnceng/internal/_build/results?buildId=676212&view=logs&j=b72e85ab-3386-5aa9-6405-3837662d9688&t=42fe3a49-1549-510e-de90-ca27b36d7602

**Comment by wtgodbe** [comment]
> /datadisks/disk1/workspace/_work/1/s/.dotnet/sdk/5.0.100-preview.6.20266.3/Microsoft.Common.CurrentVersion.targets(3035,5): error MSB3554: Cannot write to the output file "/datadisks/disk1/workspace/_work/1/s/artifacts/obj/Microsoft.AspNetCore.Razor.Language/Release/netstandard2.0/Microsoft.AspNetCore.Razor.Language.Resources.resources". The process cannot access the file '/datadisks/disk1/workspace/_work/1/s/artifacts/obj/Microsoft.AspNetCore.Razor.Language/Release/netstandard2.0/Microsoft.AspNetCore.Razor.Language.Resources.resources' because it is being used by another process. [/datadisks/disk1/workspace/_work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Language/src/Microsoft.AspNetCore.Razor.Language.csproj]

> CSC : error CS1566: Error reading resource 'Microsoft.AspNetCore.Razor.Language.Resources.resources' -- 'Could not find file '/datadisks/disk1/workspace/_work/1/s/artifacts/obj/Microsoft.AspNetCore.Razor.Language/Release/netstandard2.0/Microsoft.AspNetCore.Razor.Language.Resources.resources'.' [/datadisks/disk1/workspace/_work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Language/src/Microsoft.AspNetCore.Razor.Language.csproj]

> /datadisks/disk1/workspace/_work/1/s/src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj(81,5): error MSB3030: Could not copy the file "/datadisks/disk1/workspace/_work/1/s/artifacts/bin/Microsoft.AspNetCore.Mvc.Razor.Extensions/Release/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.dll" because it was not found.

**Comment by wtgodbe** [comment]
The same thing probably needs to be fixed in Aspnetcore-Tooling: https://dev.azure.com/dnceng/internal/_build/results?buildId=676663&view=logs&j=7c8326b9-0a5f-532a-e6de-db8515c72d9a&t=06c773de-aa2b-5fd5-6aaa-ea6cf87a3134

> ##[error]C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(124,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) The process cannot access the file 'F:\workspace\_work\1\s\artifacts\obj\Microsoft.VisualStudio.RazorExtension\Microsoft.VisualStudio.RazorExtension.csproj.nuget.dgspec.json' because it is being used by another process.

**Comment by wtgodbe** [comment]
Issue appears to be here: https://github.com/dotnet/aspnetcore/blob/c5830cf2a99b1f36ebc30c28ad16622b0d699857/src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj#L52-L57

> that sets a property ( PublishDir) and invokes Publish. Publish then does Build, which does ResolveProjectReferences--which then infects Razor.Language with the global property. that copy of Razor.Language is built in parallel with the "real" one

(Thanks @rainersigwald for helping with the investigation)

**Comment by JunTaoLuo** [comment]
@pranavkm would you be able to take a stab at fixing this issue?

**Comment by dougbu** [comment]
@pranavkm you resolved this in 0a42cca945e456bea79d6834b5b3c67cacac4875 "Razor SDK build ordering issues (#22684)", right❔
